src/lib/Env.hs

```diff
@@ -71,7 +71,7 @@ nameTag :: Name -> Tag
 nameTag name = case name of
   Name _ t _        -> t
   GlobalName t      -> t
-  NoName            -> error "NoName has no tag"
+  NoName            -> "tmp"
```

Comment:
IMO we should keep this as an error, and make the `freshVar` of each respective pass take care of assigning sensible names even for `NoName` variables, but this is fine too I guess